CMakeLists.txt

```
@@ -56,8 +56,16 @@ endif()
 
 add_compile_options(-Wall -Wextra -Werror -Wfatal-errors -Wno-missing-braces)
 
+if (NOT OTBR_VENDOR_NAME)
+    set(OTBR_VENDOR_NAME "OpenThread")
+endif()
+
+if (NOT OTBR_PRODUCT_NAME)
+    set(OTBR_PRODUCT_NAME "BorderRouter")
+endif()
+
 if(NOT OTBR_NAME)
-    set(OTBR_NAME "OPENTHREAD_BR")
+    set(OTBR_NAME "${OTBR_VENDOR_NAME}_${OTBR_PRODUCT_NAME}")
```

Thread Spec recommends name without a `_`?
```suggestion
    set(OTBR_NAME "${OTBR_VENDOR_NAME} ${OTBR_PRODUCT_NAME}")
```

---

`OTBR_NAME` is currently not used for the MeshCoP service name.

https://github.com/openthread/ot-br-posix/blob/b66e46416a34dc25ec17693affab006629db1385/src/agent/border_agent.hpp#L60-L62

I think it is using `_` because of parsing issues in test scripts. @superwhd

---

FWIW, if we introduce a space in the meshcop service instance name at registration, we will see a `\032` substring instead of a space at the discovery of the meshcop service. That's why I decided not to use a space in the service instance name.